app/models/event.rb

```diff
@@ -143,4 +145,17 @@ def self.find_current_event_id
     Event.where("start_date > ?", Date.today).order(start_date: :asc).first.id rescue nil
   end
 
+  def self.past_month(year_events, month)
+    sql = %{
+      EXTRACT(month from start_date) = ? OR
+      EXTRACT(month from end_date) = ?
+    }
+
+    year_events.where(sql, month, month).where("end_date >= ?", Date.today).blank?
```

Comment:
With a little bit of boolean logic, you could rewrite this using just one `where`, so concatenating all the statements into one SQL query.

Reply:
Thanks @Levia I just realised that I should have done this differently! I will fix this now!